include/yaramod/parser/parser_driver.h

```diff
@@ -6,38 +6,239 @@
 
 #pragma once
 
+#include <climits>
 #include <fstream>
 #include <memory>
+#include <set>
+#include <stack>
 #include <unordered_map>
 
-#include "yaramod/parser/lexer.h"
+#define FMT_HEADER_ONLY 1
+
+// Uncomment for debugging
+// #define POG_DEBUG 1
+
+#include <pog/pog.h>
+
+#include "yaramod/utils/trie.h"
+#include "yaramod/types/expressions.h"
+#include "yaramod/types/hex_string.h"
+#include "yaramod/types/hex_string.h"
+#include "yaramod/types/tokenstream.h"
+#include "yaramod/types/meta.h"
+#include "yaramod/types/plain_string.h"
+#include "yaramod/types/regexp.h"
+#include "yaramod/types/rule.h"
 #include "yaramod/types/symbol.h"
 #include "yaramod/types/yara_file.h"
-#include "yaramod/yy/yy_parser.hpp"
+#include "yaramod/yaramod_error.h"
 
 namespace yaramod {
 
-/**
- * Represents error during parsing.
- */
-class ParserError : public std::exception
+using RegexpRangePair = std::pair<std::optional<std::uint64_t>, std::optional<std::uint64_t>>;
+
+// Value is the type of all tokens produced by POG parser. Both token and rule actions return Value. The rule action parameters are also Values.
+class Value
 {
 public:
-	ParserError(const std::string& errorMsg) : _errorMsg(errorMsg) {}
-	ParserError(const ParserError&) = default;
+	using Variant = std::variant<
+		std::string, //0
+		int,
+		bool, //2
+		std::optional<TokenIt>,
+		Rule, //4
+		std::vector<Meta>,
+		std::shared_ptr<Rule::StringsTrie>, //6
+		std::pair<std::uint32_t, std::vector<TokenIt>>,
+		Literal, //8
+		Expression::Ptr,
+		std::vector<Expression::Ptr>, //10
+		std::vector<TokenIt>,
+		std::vector<std::shared_ptr<HexStringUnit>>, //12
+		std::shared_ptr<HexStringUnit>,
+		std::vector<std::shared_ptr<HexString>>, //14
+		std::shared_ptr<String>,
+		std::shared_ptr<RegexpUnit>, //16
+		std::vector<std::shared_ptr<RegexpUnit>>,
+		TokenIt, //18
+		RegexpRangePair
+	>;
+
+	/// @name Constructors
+	/// @{
+	Value(const std::string& v) : _value(v) {}
```

Comment:
Curious. Why can't there be single constructor like

```cpp
template <typename T>
Value(T&& v) : _value(std::forward<T>(v)) {}
```

?

Reply:
Actually, there can be a single constructor like 
```
template <typename T>
Value(T&& v) : _value(std::forward<T>(v)) {}
```
I fixed it.